### PR TITLE
add capability to parse dump in base64 format

### PR DESCRIPTION
**Pull request by aleks-and**
image

### CODE REVIEW

File: src/js/components/CommandPanel/CommandPanel.tsx [hunk 1]
```diff
@@ -36,6 +37,14 @@ import {THandleChange} from './types.js';
 
 const {getHexFromBytes} = joobyCodec.utils;
 
+const base64ToHex = (base64: string) => Array.from(atob(base64), char => char.charCodeAt(0).toString(16).padStart(2, '0')).join(' ');
```
Comment (DeadByElpy) [inline src/js/components/CommandPanel/CommandPanel.tsx:40]:
```suggestion
```

just use `atob`

File: src/js/components/CommandPanel/CommandPanel.tsx [hunk 2]
```diff
@@ -228,16 +242,27 @@ const CommandPanel = ({setLogs}: {setLogs: TSetLogs}) => {
             return;
         }
 
-        const hexWithoutComments = removeComments(hex);
-
+        let preparedHex;
```
Comment (DeadByElpy) [inline src/js/components/CommandPanel/CommandPanel.tsx:245]: rename origin state

[hex, setHex] to [dump, setDump]

here rename to `hex`

File: src/js/components/CommandPanel/CommandPanel.tsx [hunk 3]
```diff
@@ -379,7 +404,20 @@ const CommandPanel = ({setLogs}: {setLogs: TSetLogs}) => {
...
 
-                <Typography variant="h5">Parse message</Typography>
+                <Box sx={{display: 'flex', alignItems: 'center'}}>
+                    <Typography variant="h5">Parse message</Typography>
+                    <RadioGroup
+                        sx={{ml: 'auto'}}
+                        row
+                        aria-label="input-format"
+                        name="input-format"
+                        value={parseFormat}
```
Comment (DeadByElpy) [inline src/js/components/CommandPanel/CommandPanel.tsx:414]:
```suggestion
                        defaultValue={parseFormats.HEX}
```

File: src/js/components/CommandPanel/CommandPanel.tsx [hunk 4]
```diff
@@ -379,7 +404,20 @@ const CommandPanel = ({setLogs}: {setLogs: TSetLogs}) => {
...
                 )}
 
-                <Typography variant="h5">Parse message</Typography>
+                <Box sx={{display: 'flex', alignItems: 'center'}}>
+                    <Typography variant="h5">Parse message</Typography>
+                    <RadioGroup
+                        sx={{ml: 'auto'}}
+                        row
+                        aria-label="input-format"
+                        name="input-format"
```
Comment (DeadByElpy) [inline src/js/components/CommandPanel/CommandPanel.tsx:413]:
```suggestion
                        aria-label="dump-input-format"
                        name="dump-input-format"
```

File: src/js/components/CommandPanel/CommandPanel.tsx [hunk 5]
```diff
@@ -36,6 +37,14 @@ import {THandleChange} from './types.js';
 
 const {getHexFromBytes} = joobyCodec.utils;
 
+const base64ToHex = (base64: string) => Array.from(atob(base64), char => char.charCodeAt(0).toString(16).padStart(2, '0')).join(' ');
+
+
+const parseFormats = {
```
Comment (DeadByElpy) [inline src/js/components/CommandPanel/CommandPanel.tsx:43]:
```suggestion
const dumpInputFormats = {
```

File: src/js/components/CommandPanel/CommandPanel.tsx [hunk 6]
```diff
@@ -50,6 +59,7 @@ const CommandPanel = ({setLogs}: {setLogs: TSetLogs}) => {
     const [parameters, setParameters] = useState('');
     const [editingCommandId, setEditingCommandId] = useState<string | null>(null);
     const [recentlyEditedCommandId, setRecentlyEditedCommandId] = useState<string | null>(null);
+    const [parseFormat, setParseFormat] = useState<string>(parseFormats.HEX);
```
Comment (DeadByElpy) [inline src/js/components/CommandPanel/CommandPanel.tsx:62]:
```suggestion
    const [dumpInputFormat, setDumpInputFormat] = useState<string>(dumpInputFormats.HEX);
```

**Review by DeadByElpy — CHANGES_REQUESTED** [review]

File: src/js/components/CommandPanel/CommandPanel.tsx (hunk 1 above)
Comment (aleks-and) [reply src/js/components/CommandPanel/CommandPanel.tsx]: fixed

File: src/js/components/CommandPanel/CommandPanel.tsx (hunk 2 above)
Comment (aleks-and) [reply src/js/components/CommandPanel/CommandPanel.tsx]: done

File: src/js/components/CommandPanel/CommandPanel.tsx (hunk 4 above)
Comment (aleks-and) [reply src/js/components/CommandPanel/CommandPanel.tsx]: done

File: src/js/components/CommandPanel/CommandPanel.tsx (hunk 6 above)
Comment (aleks-and) [reply src/js/components/CommandPanel/CommandPanel.tsx]: done

**Review by DeadByElpy — APPROVED** [review]

**Pull request merged**